runtime-modules/working-team/src/benchmarking.rs

```diff
@@ -0,0 +1,599 @@
+
+      let old_stake = One::one();
+      WorkingTeam::<T, _>::decrease_stake(RawOrigin::Signed(lead_id.clone()).into(), worker_id.clone(), old_stake).unwrap();
+      let new_stake = BalanceOfCurrency::<T>::max_value();
+    }: _ (RawOrigin::Signed(caller_id.clone()), worker_id.clone(), new_stake)
+
+    // Regular worker is the worst case scenario since the checks
+    // require access to the storage whilist that's not the case with a lead opening
+    decrease_stake {
+      let i in 0 .. 10;
+
+      let (lead_id, _) = insert_a_worker::<T, I>(StakingRole::WithoutStakes, JobOpeningType::Leader, 0, None);
+      let (_, worker_id) = insert_a_worker::<T, I>(StakingRole::WithStakes, JobOpeningType::Regular, 1, Some(lead_id.clone()));
+
+      // I'm assuming that we will usually have MaxBalance > 1
+      let new_stake = One::one();
+    }: _ (RawOrigin::Signed(lead_id), worker_id, new_stake)
+
+    spend_from_budget {
+      let i in 0 .. 10;
+
+      let (lead_id, _) = insert_a_worker::<T, I>(StakingRole::WithoutStakes, JobOpeningType::Leader, 0, None);
+
+      let current_budget = BalanceOfCurrency::<T>::max_value();
+      WorkingTeam::<T, _>::set_budget(RawOrigin::Root.into(), current_budget).unwrap();
+    }: _ (RawOrigin::Signed(lead_id.clone()), lead_id.clone(), current_budget, None)
+
+    // Regular worker is the worst case scenario since the checks
+    // require access to the storage whilist that's not the case with a lead opening
+    update_reward_amount {
+      let i in 0 .. 10;
+
+      let (lead_id, _) = insert_a_worker::<T, I>(StakingRole::WithoutStakes, JobOpeningType::Leader, 0, None);
+      let (_, worker_id) = insert_a_worker::<T, I>(StakingRole::WithoutStakes, JobOpeningType::Regular, 1, Some(lead_id.clone()));
+      let new_reward = Some(BalanceOfCurrency::<T>::max_value());
+    }: _ (RawOrigin::Signed(lead_id.clone()), worker_id, new_reward)
+
+    set_status_text {
+      let i in 0 .. 50000; // TODO: We should have a bounded value for description
+
+      let (lead_id, _) = insert_a_worker::<T, I>(StakingRole::WithoutStakes, JobOpeningType::Leader, 0, None);
+      let status_text = Some(vec![0u8].repeat(i.try_into().unwrap()));
+
+    }: _ (RawOrigin::Signed(lead_id), status_text)
+
+    update_reward_account {
+      let i in 0 .. 10;
+
+      let (caller_id, worker_id) = insert_a_worker::<T, I>(StakingRole::WithoutStakes, JobOpeningType::Leader, 0, None);
+      let new_id = account::<T::AccountId>("new_id", 1, 0);
+
+    }: _ (RawOrigin::Signed(caller_id), worker_id, new_id)
+
+    set_budget {
+      let i in 0 .. 10;
+
+      let new_budget = BalanceOfCurrency::<T>::max_value();
+
+    }: _(RawOrigin::Root, new_budget)
+
+    // Regular opening is the worst case scenario since the checks
+    // require access to the storage whilist that's not the case with a lead opening
+    add_opening{
+      let i in 0 .. 50000; // TODO: We should have a bounded value for description
```

Comment:
@bedeho Despite the fact we don't use the text fields in the runtime because of the query node - we need to bound the length of these values in order to properly benchmark such processes like text hashing.

Reply:
I think there is a way to do this without needing to bound it, I believe the extrinsic caller needs to commit to a length. Please elaborate @conectado.

Reply:
That way of doing things works for weights that depend on an stored item.

Say, for example, an extrinsic that depends on the number of stored workers, then instead of assuming worst-case scenario(that there are max number of workers stored) you can ask the caller to pass the number of workers as a parameter and then ensure that it is correct before executing the code. That way you can use that parameter to calculate the weight instead of assuming worst-case and refunding later.

Here, it is not the case, because we depend directly on the parameter passed, the problem is that this parameter is unbounded and for what I've seen in FRAME is that they actually bound with a "max" value all these linear parameters (like [here](https://github.com/paritytech/substrate/blob/900d034fc1ba0ccb30dd54b2dd7498ad5caa7fb5/frame/treasury/src/lib.rs#L633)).

However, I don't fully understand why this is needed. What I understand is that since we are doing linear interpolation, if the extrinsic (meaning, in this case, the hash function) is really linear we can use an arbitrarily large value as an upper-bound only for the linear interpolation and then depend on the Weight of the extrinsic call to bound the function running-time. But I've not seen it done like that anywhere.

Reply:
Ok. Let's skip the upper limits and rely on block length fees and linear dependency on the text length.